src/Build/CompatibilitySuppressions.xml

```diff
@@ -58,4 +58,173 @@
     <Right>ref/net8.0/Microsoft.Build.dll</Right>
     <IsBaselineSuppression>true</IsBaselineSuppression>
   </Suppression>
-</Suppressions>
+  <!-- Suppressions for api differences introduced in the new binlog reading events -->
```

Comment:
Are we fine with those?
Caused by moving the methods of more granular interface into the `IBuildEventArgsReaderNotifications`
more details: https://github.com/dotnet/msbuild/pull/9307#discussion_r1391703521

The fact that those were not flagged when added to the interface via inheritance feels as a deficiency of the api compat check anyway (and it's actually tracked: https://github.com/dotnet/sdk/issues/25888)

We are extending public interface, but that one is not meant by public implementation, only consumption. Quick GH search yields that there is no implementation of that interface beyond msbuild codebase.

**tl;dr;:** I'm quite opinionated that we are fine here.

@rainersigwald - would you disagree?

Btw. it feels very tempting to get rid of `StringEncountered` at the same time - it as well is not used in any public code. The binlogs strings and files reading feature is not yet fully deployed - so the usefulness up till now would be limited and questionable.

Reply:
I . . . don't know! This feels like a very small breaking change that we could get away with, since it's not like we announced or publicized anything and as you say there's no reason to inherit from this interface.

What would it take to avoid this?

Reply:
> What would it take to avoid this?

* Keeping the granular interfaces (IBuildEventStringsReader, IBuildFileReader, IBinaryLogReaderErrors) - which now, looking back, I admit it's really unnecessary level of detail for API user
* Everywhere wehre we implement `IBuildEventArgsReaderNotifications` (so BinlogReader, ReplaySource), we'd need to explicitly implement those granular interfaces as well

**tl;dr;** - we'd have unnecesary bloat of API, plus the `IBuildEventArgsReaderNotifications` would became confusing as it'd contain only the deduplicated string notifications, but no notifications about embedded files, not errors during reading.

It'd be nice to roll out the API more clean and less confusing

Reply:
My opinion is we should fearlessly and aggressively change public API introduced relatively recently as we see fit instead of keeping our design experiments for the next 30 years. Most users of MSBuild I know are on 17.5 if I'm being generous, and those that are on the bleeding edge are more likely to adapt in the infinitesimal chance they took a dependency already. At the very least if a third-party is using any of this, I sure would like to know!